### ISSUE TITLE
Static type conversion with Bool

### ISSUE THREAD

**Issue by oschulz** [SciML/Static.jl#91]
We currently have

```julia
julia> promote(static(5), static(true))
(5, 1)

julia> promote(static(true), true)
(static(true), static(true))

julia> promote(static(true), static(true))
(static(true), static(true))

julia> promote(static(true), true)
(static(true), static(true))
```

Is this deliberate, or some unintended effect? It seems weird that we promote static integer and static bool to non-static, and promote static integer and non-static bool to static.

**Comment by chriselrod** [comment]
Bug.

**Comment by Tokazama** [comment]
Fixed with #110
